### ISSUE TITLE
cryptopp-project error on build on xcode

### ISSUE THREAD

**Issue by fernandoamorim** [ethereum/cpp-ethereum#3622]
Hello!

I have an error on build the project:

<img width="992" alt="screen shot 2017-03-07 at 03 30 23" src="https://cloud.githubusercontent.com/assets/10285234/23644507/6c8f4330-02e6-11e7-91f8-7b804a11398c.png">

**Comment by chfast** [comment]
The issue is CMake cannot find CMake :) Do you have the `cmake` executable hidden somewhere?

I think I know how to fix it, will include required change in https://github.com/ethereum/cpp-ethereum/pull/3610, but it may take some time to be merged in.

As a workaround, can you add `cmake` executable to the PATH environment variable?

**Comment by fernandoamorim** [comment]
chfast,

I have CMake.app in my Applications folder (like img).
And when i execute make os cmake in terminar working.

What i need make to fix it?
<img width="135" alt="screen shot 2017-03-07 at 09 47 06" src="https://cloud.githubusercontent.com/assets/10285234/23657054/0b403d2a-031b-11e7-959d-cb69ca192038.png">

**Comment by chfast** [comment]
In file `deps/cryptopp.cmake` (https://github.com/ethereum/cpp-dependencies/blob/ffb582905800966cf15471a852b8842bafde0aa7/cryptopp.cmake#L24) change in line 24: `cmake -E` to `${CMAKE_COMMAND} -E`.

If that will move you further let me know.

**Comment by fernandoamorim** [comment]
I change line 24. But still not working (same problem).

I need create a project again?

mkdir build_xc
cd build_xc
cmake -G Xcode ..

**Comment by chfast** [comment]
It might be enough to just reload cmake doing `cmake .` in `build_xc` dir.

**Comment by fernandoamorim** [comment]
I try cmake -G Xcode .. again, but still not wotking.. :(

<img width="982" alt="screen shot 2017-03-07 at 10 36 00" src="https://cloud.githubusercontent.com/assets/10285234/23658589/f2afaff0-0321-11e7-8869-3fc5a0b49847.png">

**Comment by chfast** [comment]
You have removed `remove` command after `-E`. Must be `${CMAKE_COMMAND} -E remove`.

**Comment by fernandoamorim** [comment]
Progress!!!

But i have same error in jsoncpp-project-stamp :(

<img width="994" alt="screen shot 2017-03-07 at 10 51 00" src="https://cloud.githubusercontent.com/assets/10285234/23659175/0ec5c1be-0324-11e7-97e3-a660fdac10ca.png">

**Comment by chfast** [comment]
Yes, there are other raw `cmake` used on other places in form of `cmake --build <BINARY_DIR>`. That also has to be changed to `${CMAKE_COMMAND} --build <BINARY_DIR>`.

They are in jsoncpp.cmake, jsonrpc.cmake and cryptopp.cmake files. That should be all.

**Comment by fernandoamorim** [comment]
I change in 2 files:

jsonrpc.cmake
BUILD_COMMAND ${CMAKE_COMMAND} --build <BINARY_DIR> --config Release
INSTALL_COMMAND ${CMAKE_COMMAND} --build <BINARY_DIR> --config Release --target install

jsoncpp.cmake
BUILD_COMMAND ${CMAKE_COMMAND} --build <BINARY_DIR> --config Release
INSTALL_COMMAND ${CMAKE_COMMAND} --build <BINARY_DIR> --config Release --target install


But still same error, but now in jsonrpc (not in jsoncpp)

**Comment by chfast** [comment]
Try to reload cmake project as before. If that does not help, delete `build_xc/deps` directory.

**Comment by fernandoamorim** [comment]
I remove the build_xc folder and make "cmake -G Xcode .." again.

But, still not working! Fu..! :(

**Comment by chfast** [comment]
Can you send me error message?

**Comment by fernandoamorim** [comment]
Ops. Progress!

But, new error :p

<img width="996" alt="screen shot 2017-03-07 at 11 23 47" src="https://cloud.githubusercontent.com/assets/10285234/23660556/9ad96080-0328-11e7-802a-f98eed8bd8b3.png">

**Comment by chfast** [comment]
I'm not sure what this one is about...

**Comment by fernandoamorim** [comment]
I follw this article:

http://www.ethdocs.org/en/latest/ethereum-clients/cpp-ethereum/building-from-source/macos.html

I will remove all cpp folder and instal via brew: brew install cpp-ethereum

**Comment by chfast** [comment]
Have you tried if compilation with `make` works? I have no experience with Xcode to be honest.

**Comment by fernandoamorim** [comment]
Or i remove all cpp folder and install again...

**Comment by fernandoamorim** [comment]
I remove and create all again.. but same error

<img width="989" alt="screen shot 2017-03-07 at 12 25 25" src="https://cloud.githubusercontent.com/assets/10285234/23663378/3020c87e-0331-11e7-8bf3-9089fed1e117.png">

**Comment by chfast** [comment]
@pirapira can you help here? Do you use XCode like in this issue?

@fernandoamorim I suggested to try not to use XCode, just pure `make` to compile it (instructions are on the same page in documentation you referenced).

**Comment by fernandoamorim** [comment]
make install?  i execute this.

**Comment by chfast** [comment]
@fernandoamorim I mean: don't generate Xcode project and open Xcode at all. Just `cmake ..` and `make`.

**Comment by fernandoamorim** [comment]
But how i start mining? 

I`m new in this things, sorry!

**Comment by pirapira** [comment]
@fernandoamorim you have Terminal?

**Comment by fernandoamorim** [comment]
Yes @pirapira

**Comment by fernandoamorim** [comment]
I follow this steps in terminal: 

http://www.ethdocs.org/en/latest/ethereum-clients/cpp-ethereum/building-from-source/macos.html

But when i build app, have the erros...

**Comment by pirapira** [comment]
Then, `cd` into the directory where you cloned `cpp-ethereum`, and then
```
mkdir -p build
cd build
cmake ..
make
```

should compile something.

**Comment by fernandoamorim** [comment]
@pirapira 
Already exists an _build_ folder into cpp-ethereum. I remove and create again?

**Comment by pirapira** [comment]
Any new directory should work.

**Comment by fernandoamorim** [comment]
@pirapira : running make.

After make command. What i need do?

**Comment by pirapira** [comment]
Previous errors on this thread do not seem like produced in the Terminal app.  What kind of errors do you get in Terminal?

**Comment by pirapira** [comment]
If `make` finishes successfully, you can type `make install` if you'd like to.

**Comment by fernandoamorim** [comment]
@pirapira : 30% and no errors yet.

**Comment by fernandoamorim** [comment]
[ 70%] Building CXX object eth/CMakeFiles/eth.dir/main.cpp.o
make[2]: *** No rule to make target `../deps/boost/stage/lib/libboost_thread.a', needed by `eth/eth'.  Stop.
make[1]: *** [eth/CMakeFiles/eth.dir/all] Error 2
make: *** [all] Error 2

**Comment by pirapira** [comment]
OK.  I made a mistake.  Go back to `..` and type
```
git submodule init
git submodule update
```
and then go back to `build` (or your new directory) and `make` again.

**Comment by fernandoamorim** [comment]
@pirapira running

**Comment by fernandoamorim** [comment]
Same error:

[ 70%] Building CXX object eth/CMakeFiles/eth.dir/main.cpp.o
make[2]: *** No rule to make target `../deps/boost/stage/lib/libboost_thread.a', needed by `eth/eth'.  Stop.
make[1]: *** [eth/CMakeFiles/eth.dir/all] Error 2
make: *** [all] Error 2

**Comment by pirapira** [comment]
What happens with `ls ../deps`?

What does `git show | head -n 4` say?

**Comment by fernandoamorim** [comment]
<img width="1082" alt="screen shot 2017-03-07 at 18 46 19" src="https://cloud.githubusercontent.com/assets/10285234/23679337/67c13df6-0366-11e7-9807-5db7f1cd290f.png">

<img width="463" alt="screen shot 2017-03-07 at 18 47 04" src="https://cloud.githubusercontent.com/assets/10285234/23679381/80c9cf02-0366-11e7-8fd9-c09e9cd125bb.png">

**Comment by pirapira** [comment]
I'll try on my machine.

One thing you can try meanwhile is to follow the script Travis runs on every pull-request: https://github.com/ethereum/cpp-ethereum/blob/develop/.travis.yml#L107

**Comment by fernandoamorim** [comment]
You said execute this script?
./scripts/install_deps.sh

i execute and return -bash: ./scripts/install_deps.sh: No such file or directory

**Comment by pirapira** [comment]
No, but you can try to do the same thing as [what Travis does on a Mac OS](https://travis-ci.org/ethereum/cpp-ethereum/jobs/208527613).

**Comment by chfast** [comment]
I don't want to break this fun, but @fernandoamorim you will not be able to do GPU mining with latest cpp-ethereum. You probably should checkout some community-driven solutions for mining. E.g. https://github.com/Genoil/cpp-ethereum.

**Comment by fernandoamorim** [comment]
I running this steps.. https://travis-ci.org/ethereum/cpp-ethereum/jobs/208527613

What the difference between two?

**Comment by fernandoamorim** [comment]
executing from JOB, > 70% and not erros .....

**Comment by fernandoamorim** [comment]
100% and no errors!!!

<img width="585" alt="screen shot 2017-03-07 at 19 34 18" src="https://cloud.githubusercontent.com/assets/10285234/23681257/1a1cd6ee-036d-11e7-9764-800373fabc96.png">

**Comment by fernandoamorim** [comment]
So when i execute cd $TRAVIS_BUILD_DIR/build && ../scripts/tests.sh $TRAVIS_TESTS return:

-bash: cd: /build: No such file or directory

**Comment by pirapira** [comment]
That's just because `TRAVIS_BUILD_DIR` is not defined.

By the way, I think @chfast is right about GPU mining.

**Comment by vandabond** [comment]
Hello guys I don't know if anybody here has any idea on how ethereum propose to help someone with ideas to decentralize the currency in his or way. I have an idea I would love some developers in the house to join me execute in my country with the help of ethereum thou.

**Comment by pirapira** [comment]
@vandabond your comment is out of topic here in this issue.
